src/EditorFeatures/Test2/Diagnostics/DiagnosticServiceTests.vb

```diff
@@ -594,6 +595,11 @@ class AnonymousFunctions
 
             Using workspace = TestWorkspaceFactory.CreateWorkspace(test)
                 Dim project = workspace.CurrentSolution.Projects.Single()
+
+                ' turn off heuristic
+                Dim options = workspace.Services.GetService(Of IOptionService)()
+                options.SetOptions(options.GetOptions.WithChangedOption(InternalDiagnosticsOptions.UseCompilationEndCodeFixHueristic, False))
```

Comment:
heuristic is that, if there is existing compilation end diagnostic, we show code fix for it, otherwise, we won't. 

but this behavior is controlled by an option you can change from Tools/Options/Roslyn/Diagnostics

until we have async support from platform, we probably need to test several different experiences. this is just one of them.